gfop/foodflows.py

```diff
@@ -0,0 +1,122 @@
+# Standard library imports
+import os
+import pkg_resources
+from importlib import resources
+from typing import List, Tuple, Optional
+
+# Third-party imports
+import numpy as np
+import pandas as pd
+
+# Internal imports
+from utils import _load_food_metadata, _load_sample_types, _validate_groups
+
+
+class FoodFlows:
+    def __init__(
+        self,
+        gnps_network: str,
+        sample_types: str,
+        groups_included: List[str],
+        max_hierarchy_level: int = 6,
+    ) -> None:
+        """
+        Initializes the FoodFlows object and creates the flows and processes
+        dataframes needed to visualize the flow of foods in the network.
+
+        Parameters
+        ----------
+        gnps_network : str
+            Path to the TSV file generated from classical molecular networking.
+        sample_types : str
+            One of 'simple', 'complex', or 'all'.
+        groups_included : List[str]
+            Groups of interest in the molecular network to generate food flows.
+        max_hierarchy_level : int, optional
+            Maximum level for food flows calculation, by default 6.
+        """
+        # Load GNPS network data
+        self.gnps_network = pd.read_csv(gnps_network, sep="\t")
+        self.food_metadata = _load_food_metadata()  # Call the utility function
+        self.sample_types = _load_sample_types(
+            self.food_metadata, sample_types
+        )  # Call the utility function
+        self.groups_included = groups_included
+        self.max_hierarchy_level = max_hierarchy_level
+
+        # Validate group names
+        _validate_groups(
+            self.gnps_network, self.groups_included
+        )  # Call the utility function
+
+        # Generate flows and processes dataframes
+        self.flows, self.processes = self.generate_foodflows()
+
+    def generate_foodflows(self) -> Tuple[pd.DataFrame, pd.DataFrame]:
+        """
+        Generates food flows and processes from the GNPS network data.
+
+        Returns
+        -------
+        Tuple[pd.DataFrame, pd.DataFrame]
+            A tuple containing:
+            - flows: DataFrame with source, target, and value columns.
+            - processes: DataFrame with process details.
+        """
+        # Select GNPS job groups.
```

Comment:
**question:** This first part repeats some behavior of the `FoodCounts` class. Can `FoodFlows` instead take a `FoodCounts` object as input on which these steps are already applied, and then only take care of the visualization?